lib/puppet/catalog-diff/searchfacts.rb

```diff
@@ -21,14 +21,14 @@ def find_nodes(options = {})
       active_nodes
     end
 
-    def find_nodes_puppetdb(env)
-      require 'puppet/util/puppetdb'
-      server_url = Puppet::Util::Puppetdb.config.server_urls[0]
-      port = server_url.port
-      use_ssl = port != 8080
-      connection = Puppet::Network::HttpPool.http_instance(server_url.host, port, use_ssl)
+    def build_query(env, version)
       base_query = ['and', ['=', ['node', 'active'], true]]
-      base_query.concat([['=', 'catalog_environment', env]]) if env
+      if version == 'latest'
```

Comment:
Or maybe a compare version with a minimal known value, to avoid edge cases in the future?

Reply:
Also, this could probably use a ternary operator.

Reply:
I just rebased my branch with the latest from master and implemented the PuppetDB version >= 3 as you suggested using a ternary operator so this should be solved now.